### ISSUE TITLE
False negative about the rule NP_METHOD_PARAMETER_TIGHTENS_ANNOTATION

### ISSUE THREAD

**Issue by LynnBroe** [spotbugs/spotbugs#2557]
Version: 4.7.3

I found a false negative about the rule [NP_METHOD_PARAMETER_TIGHTENS_ANNOTATION](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#np-method-tightens-nullness-annotation-on-parameter-np-method-parameter-tightens-annotation).

In the example below, SpotBugs reports a warning at line 9. The 'test' method in class A takes a parameter that is marked as @Nullable, and Class B overrides that method with a parameter is @Nonnull.

```java
public class Test {
    public class A{
        public Object test(@org.jetbrains.annotations.Nullable Object a){
            return a;
        }
    }
    public class B extends A{
        public Object test(@javax.annotation.Nonnull Object a){
            return a; // can report a warning
        }
    }
}
```

However, in the example below, SpotBugs reports no warning if we use `@javax.annotation.Nullable` instead. Actually, `@org.jetbrains.annotations.Nullable` and `@javax.annotation.Nullable` are equivalent. So, this exmaple also violates the rule NP_METHOD_PARAMETER_TIGHTENS_ANNOTATION and SpotBugs should report a warning at line 9.

```java
public class Test {
    public class A {
        public Object test(@javax.annotation.Nullable Object a){
            return a;
        }
    }
    public class B extends A {
        public Object test(@javax.annotation.Nonnull Object a){
            return a;  //  should report a warning, but no warning
        }
    }
}
```

Hence, I think this is a false negative.

**Comment by LynnBroe** [comment]
I found that SpotBugs reports no warning if we use `@edu.umd.cs.findbugs.annotations.Nullable` instead. 
```java
public class Test {
    public class A {
        public Object test(@edu.umd.cs.findbugs.annotations.Nullable Object a){
            return a;
        }
    }
    public class B extends A {
        public Object test(@javax.annotation.Nonnull Object a){
            return a;  //  should report a warning, but no warning
        }
    }
}
```
